IT-Wizards-API/src/main/java/org/LaunchCode/IT_Wizards_API/services/CartService.java

```diff
@@ -0,0 +1,94 @@
+package org.LaunchCode.IT_Wizards_API.services;
+
+import jakarta.transaction.Transactional;
+import org.LaunchCode.IT_Wizards_API.exceptions.CartItemNotFoundException;
+import org.LaunchCode.IT_Wizards_API.exceptions.CartNotFoundException;
+import org.LaunchCode.IT_Wizards_API.exceptions.OrdersNotFoundException;
+import org.LaunchCode.IT_Wizards_API.exceptions.UserNotFoundException;
+import org.LaunchCode.IT_Wizards_API.models.*;
+import org.LaunchCode.IT_Wizards_API.repository.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CartService {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private CartItemRepository cartItemRepository;
+
+    @Autowired
+    private ItemRepository itemRepository;
+
+    public CartItem addItemToCart(Long userId, Long newItemId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new UserNotFoundException(userId));
+        Cart cart;
+        CartItem cartItem;
+       try {
+            cart = cartRepository.findByUserId(userId).orElseThrow(() -> new CartNotFoundException(userId));
+       } catch(Exception e) {
+           cart = cartRepository.save(new Cart(user));
+        }
+        Item item = itemRepository.findById(newItemId)
+                .orElseThrow(() -> new IllegalArgumentException("Item not found"));
+        //TODO: Look if cartItem exists & if so up quantity by 1
```

Comment:
This todo line can be removed now, right?

Reply:
Yes, I forgot to clear that Todo, I will do so on my next commit along with useNavigate in CartContext.